packages/ag-charts-community/src/chart/series/cartesian/barSeries.ts

```diff
@@ -252,67 +277,147 @@ export class BarSeries extends AbstractBarSeries<Rect, BarSeriesProperties, BarN
+                height: barAlongX ? barWidth * crossScale : Math.abs(bboxBottom - bboxHeight),
+            };
+
+            const {
+                fontStyle: labelFontStyle,
+                fontWeight: labelFontWeight,
+                fontSize: labelFontSize,
+                fontFamily: labelFontFamily,
+                color: labelColor,
+                placement,
+            } = label;
+            const lengthRatioMultiplier = this.shouldFlipXY() ? rect.height : rect.width;
+
+            return {
+                series: this,
+                itemId: yKey,
+                datum,
+                valueIndex,
+                cumulativeValue,
+                xValue,
+                yValue,
+                yKey,
+                xKey,
+                capDefaults: {
+                    lengthRatioMultiplier: lengthRatioMultiplier,
+                    lengthMax: lengthRatioMultiplier,
+                },
+                x: barRect.x,
+                y: barRect.y,
+                width: barRect.width,
+                height: barRect.height,
+                midPoint: { x: rect.x + rect.width / 2, y: rect.y + rect.height / 2 },
+                fill,
+                stroke,
+                opacity: 1,
+                strokeWidth,
+                cornerRadius,
+                topLeftCornerRadius: barAlongX !== isUpward,
+                topRightCornerRadius: isUpward,
+                bottomRightCornerRadius: barAlongX === isUpward,
+                bottomLeftCornerRadius: !isUpward,
+                clipBBox,
+                label:
+                    labelText != null
+                        ? {
+                              text: labelText,
+                              fill: labelColor,
+                              fontStyle: labelFontStyle,
+                              fontWeight: labelFontWeight,
+                              fontSize: labelFontSize,
+                              fontFamily: labelFontFamily,
+                              ...adjustLabelPlacement({
+                                  isPositive,
+                                  isVertical: !barAlongX,
+                                  placement,
+                                  rect,
+                              }),
+                          }
+                        : undefined,
+                missing: yValue == null,
+            };
+        };
+
         const { groupScale, processedData } = this;
         processedData?.data.forEach(({ keys, datum: seriesDatum, values, aggValues }) => {
             values.forEach((value, valueIndex) => {
                 const xValue = keys[xIndex];
-                const x = xScale.convert(xValue);
-
-                const currY = +value[yEndIndex];
-                const prevY = +value[yStartIndex];
                 const yRawValue = value[yRawIndex];
+                const yStart = +value[yStartIndex];
```

Comment:
```suggestion
                const yStart = Number(value[yStartIndex]);
```
nit: I'd prefer that we avoid these sort of hacks as they make the code less readable IMO.